client/app/lib/maincontroller/groupscontroller.coffee

```diff
@@ -42,6 +44,19 @@ module.exports = class GroupsController extends kd.Controller
       @on 'InstanceDeleted', (data) ->
         remote_extensions.removeInstance data?.contents
 
+      @on 'MembershipRoleChanged', (data) ->
+
+        { reactor } = kd.singletons
+        { contents: { role, id, adminNick } } = data
+        reactor.dispatch 'UPDATE_TEAM_MEMBER_WITH_ID', { id, role }
+
+        if id is whoami()._id
+          modal = new MembershipRoleChangedModal
+            success: ->
+              modal.destroy()
+              global.location.reload yes
+          , { role, adminNick }
+
```

Comment:
`GroupsController` isn't the place where we should do such things. I'd prefer this to be in either `MainView` or `MainViewController`. wdyt @gokmen ?